packages/api/src/providers/fitbit.ts

```diff
@@ -359,21 +360,41 @@ export class Fitbit extends Provider implements OAuth2 {
       [FitbitScopes.weight]: "body",
     };
 
-    const subscriptionId = userId;
-
+    const subscriptionId = fitbitUserId;
+    const activeSubscriptions = await this.checkExistingSubscriptions(accessToken);
```

Comment:
so the logic generally LGTM - but there's one oversight:

if there's already existing subs for that user - it means that another fitbit user in our DB already exists, which also means that in future webhook requests, [this](https://github.com/metriport/metriport/blob/9b8e1d4d469a07ef3db57aa25cb63616c4f710da/packages/api/src/command/webhook/fitbit.ts#L38-L41
) may find the wrong user.

suggestion:
- if there are active subs, find the existing connected user with that fitbit id, and [revoke the token locally](https://github.com/metriport/metriport/blob/9b8e1d4d469a07ef3db57aa25cb63616c4f710da/packages/api/src/providers/oauth2.ts#L168-L174)
- then, [send the cx a wh saying that the old user has been disconnected](https://github.com/metriport/metriport/blob/9b8e1d4d469a07ef3db57aa25cb63616c4f710da/packages/api/src/command/webhook/devices.ts#L68) from fitbit (because they have)

Reply:
Fixed. Also added re-subscriptions for WH notifications.